pkg/controller/jobframework/reconciler.go

```diff
@@ -328,6 +372,22 @@ func (r *JobReconciler) ensureOneWorkload(ctx context.Context, job GenericJob, o
 		}
 	}
 
+	// the job and workload might not be equivalent if the previous
+	// reconcile failed mid stopJob, in which case the job is IsSuspended
+	// the workload is not Admitted but it has the finalizer set
+	// search again
+	if match == nil && job.IsSuspended() {
```

Comment:
Similarly here, is this needed after https://github.com/kubernetes-sigs/kueue/pull/1005?